CMakeLists.txt

```diff
@@ -326,108 +326,124 @@ function(determine_repository_version source_dir build_repository have_build_rep
-endif ()
+endif()
 
 ################################################################################
 ## OPERATING SYSTEM
 ################################################################################
 
 if(CMAKE_SYSTEM_NAME MATCHES ".*Linux")
   set(LINUX TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "kFreeBSD.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "kFreeBSD.*")
   set(FREEBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "kNetBSD.*|NetBSD.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "kNetBSD.*|NetBSD.*")
   set(NETBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "kOpenBSD.*|OpenBSD.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "kOpenBSD.*|OpenBSD.*")
   set(OPENBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES ".*GNU.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES ".*GNU.*")
   set(GNU TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES ".*BSDI.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES ".*BSDI.*")
   set(BSDI TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "DragonFly.*|FreeBSD")
+elseif(CMAKE_SYSTEM_NAME MATCHES "DragonFly.*|FreeBSD")
   set(FREEBSD TRUE)
-elseif (CMAKE_SYSTEM_NAME MATCHES "SYSV5.*")
+elseif(CMAKE_SYSTEM_NAME MATCHES "SYSV5.*")
   set(SYSV5 TRUE)
-endif ()
+endif()
+
+# ------------------------------------------------------------------------------
+# LINUX system call GETTID
+# ------------------------------------------------------------------------------
+
+option(CHECK_GETTID "check for the existance of gettid" ON)
+
+if(CHECK_GETTID)
+  message(STATUS "Checking for gettid, use '-DCHECK_GETTID=Off' if cross compiling")
+  try_compile(GETTID_COMPILE_TEST
+          SOURCE_FROM_FILE "check_tid_compilex.c" ${CMAKE_SOURCE_DIR}/cmake/check_tid_compile.c
+          OUTPUT_VARIABLE GETTID_COMPILE_RESULT
+  )
+
+  if(GETTID_COMPILE_TEST)
+      message(STATUS "gettid appears to be available at compile time.")
+      add_definitions("-DHAVE_SYS_GETTID=1")
+  else()
+      message(STATUS "gettid does NOT appear to be available at compile time.")
+  endif()
+endif()
 
 # ------------------------------------------------------------------------------
 # user options
 # ------------------------------------------------------------------------------
 
-option(
-  USE_JEMALLOC
-  "use jemalloc memory allocator"
-  ON
-)
+option(USE_JEMALLOC "use jemalloc memory allocator" ON)
 
-option(
-  USE_LIBUNWIND
-  "use libunwind for stack traces"
-  ON
-)
+option(USE_LIBUNWIND "use libunwind for stack traces" ON)
 
 option(USE_BUILD_ID_READER "Add code to read arangod's build-id from within arangod" ON)
 
-if (USE_BUILD_ID_READER)
+if(USE_BUILD_ID_READER)
   add_definitions(-DUSE_BUILD_ID_READER=true)
-else ()
+else()
   add_definitions(-DUSE_BUILD_ID_READER=false)
-endif ()
+endif()
 
 # Guess whether we're using mold
+set(mold_test_flags " --version ${CMAKE_EXE_LINKER_FLAGS}")
+string(REPLACE " " " -Wl," mold_test_flags "${mold_test_flags}")
+
 execute_process(
-  COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_EXE_LINKER_FLAGS} -Wl,--version
+  COMMAND "${CMAKE_CXX_COMPILER} ${mold_test_flags}"
```

Comment:
I don't get why you did this change, you use the variable `mold_test_flags` nowhere else (so it is not really needed), I think it was much better readable before your change.

Reply:
@jvolmer there is an issue with the original code: Assume that CMAKE_EXE_LINKER_FLAGS contains more than one flag, ie "-pthread -fopenmp" than this will be passed as ONE argument, instead of TWO. The result is 

    clang++-19: error: unknown argument: '-pthread -fopenmp'

I will add a comment